libx509pq.c

```diff
@@ -638,6 +639,130 @@ Datum x509_publickeymd5(
 	PG_RETURN_NULL();
 }
 
+PG_FUNCTION_INFO_V1(x509_tbscert_strip_sct);
```

Comment:
Is it necessary to have x509_tbscert_strip_sct() and x509_tbscert_strip_poison() as separate functions?

I think I'd prefer to have just one function called x509_tbscert_stripCTExtensions() that will delete whichever CT extension(s) are present in the cert and then return the tbscert.  Would that work for you?